### PR TITLE
Add some features that TreeCorr uses.

### PR DESCRIPTION
**Pull request by rmjarvis**
I have had a similar set of classes in TreeCorr for various coordinate calculations I do there.  I'd like to transition over to using Coord and remove the TreeCorr versions of these, but there are a few things I need that aren't in the current Coord package:

* `xyz_to_radec` converts from 3D x,y,z coordinates to ra, dec in radians.  Specifically, this needs to be able to handle numpy arrays.  Other than that, it is equivalent to
```
coord = CelestialCoord.from_xyz(x, y, z)
ra, dec = coord.ra.rad, coord.dec.rad
```
* `radec_to_xyz` converts from ra, dec coordinates in radians to x,y,z.  Again, accepting numpy arrays.  It is otherwise equivalent to
```
x,y,z = CelestialCoord(ra * radians, dec * radians).get_xyz()
```
* `AngleUnit.valid_names` lists the names that are valid inputs to `AngleUnit.from_name`.
* Along the way I fixed @erykoff 's bug report #8.

I don't know if anyone will want to code review this.  But I'll plan to merge this Friday unless someone speaks up.

### CODE REVIEW

File: coord/celestial.py [hunk 1]
```diff
@@ -209,6 +209,82 @@ def from_xyz(x, y, z):
...
+
+        :param ra:      The right ascension(s) in radians. May be a numpy array.
+        :param dec:     The declination(s) in radians. May be a numpy array.
+        :param r:       The distance(s) from Earth (default 1.). May be a numpy array.
+
+        :returns: x, y, z as a tuple.
+        """
+        import numpy
+        cosdec = numpy.cos(dec)
+        x = cosdec * numpy.cos(ra) * r
```
Comment (jmeyers314) [inline coord/celestial.py]: Maybe a good opportunity to make a vectorized sincos?

File: coord/celestial.py [hunk 2]
```diff
@@ -209,6 +209,82 @@ def from_xyz(x, y, z):
...
+
+        :param x:       The x position(s) in 3 dimensions. May be a numpy array.
+        :param y:       The y position(s) in 3 dimensions. May be a numpy array.
+        :param z:       The z position(s) in 3 dimensions. May be a numpy array.
+
+        :returns: ra, dec as a tuple.
+        """
+        import numpy
+        ra = numpy.arctan2(y, x)
+        dec = numpy.arctan2(z, numpy.sqrt(x**2+y**2))
```
Comment (jmeyers314) [inline coord/celestial.py]: This line's about 10x faster for me as `arctan(z/np.sqrt(x**2+y**2))`, which I think is okay.  (We already know the right quadrants.)

File: coord/celestial.py [hunk 3]
```diff
@@ -209,6 +209,82 @@ def from_xyz(x, y, z):
...
+        return ra,dec
```
Comment (jmeyers314) [inline coord/celestial.py]: It's about 10% more expensive here to also calculate `r=np.sqrt(x**2+y**2+z**2)`.  Maybe make that at least an optional output for symmetry?

**Review by jmeyers314 — APPROVED** [review]
LGTM.  I listed a few small ideas for you to look at.

**Comment by rmjarvis** [comment]
Thanks Josh.  I didn't make a vectorized sincos.  But I took your other two suggestions.

Some grist for the sincos mill:
https://stackoverflow.com/questions/32397347/is-there-a-fast-way-to-return-sin-and-cos-of-the-same-value-in-python
https://github.com/numpy/numpy/issues/2626

I think the upshot is that to do this, we'd have to roll our own in C++ that can call MKL's `vdSinCos` if it's available.  That's more work than I want to do right now.  (Mostly in setup.py to find the MKL library if it's installed and have a backup plan if not.)

**Pull request merged**